lib/wallet/wallet.js

```diff
@@ -2569,6 +2570,157 @@ class Wallet extends EventEmitter {
+    const ns = await this.getNameState(nameHash);
+    const height = this.wdb.height + 1;
+    const network = this.network;
+
+    if (!ns)
+      throw new Error(`Auction not found: "${name}".`);
+
+    if (ns.isExpired(height, network))
+      throw new Error(`Name has expired: "${name}"!`);
+
+    const state = ns.state(height, network);
+
+    if (state < states.CLOSED)
+      throw new Error(`Auction is not yet closed: "${name}".`);
+
+    const reveals = await this.txdb.getReveals(nameHash);
+    const outpoints = [];// { outpoint, output }
+
+    for (const {prevout, own} of reveals) {
+      const {hash, index} = prevout;
+
+      if (!own)
+        continue;
+
+      const coin = await this.getCoin(hash, index);
+
+      if (!coin)
+        continue;
+
+      if (acct != null && !await this.txdb.hasCoinByAccount(acct, hash, index))
+        continue;
+
+      // Is local?
+      if (coin.height < ns.height)
+        continue;
+
+      const output = new Output();
+      output.address = coin.address;
+      output.value = coin.value;
+      output.covenant.type = types.REDEEM;
+      output.covenant.pushHash(nameHash);
+      output.covenant.pushU32(ns.height);
+
+      if (prevout.equals(ns.owner)) { // Winner, does REGISTER
+        output.covenant.type = types.REGISTER;
+        // second highest bid
+        output.value = ns.value;
+        const raw = resource.encode();
+        if (raw.length > rules.MAX_RESOURCE_SIZE)
+            throw new Error(`Resource exceeds maximum size: "${name}".`);
+        output.covenant.push(raw);
+        output.covenant.pushHash(await this.wdb.getRenewalBlock());
+      }
+      outpoints.push({ output: output, outpoint: prevout });
+    }
+
+    return outpoints;
+  }
+
+    /**
+   * Make a batch finish MTX.
+   * @param {Array<{name: String, Data: Array}>} names
+   * @param {(Number|String)} acct
+   * @returns { { MTX, errorMessages:[name: String, errorMessage: String] } }
+   */
+
+  async makeBatchFinish(names, acct) {
+    assert(Array.isArray(names));
+
+    if (acct != null) {
+      assert((acct >>> 0) === acct || typeof acct === 'string');
+      acct = await this.getAccountIndex(acct);
+    }
+
+    const mtx = new MTX();
+    const errorMessages = [];
+    const MAX_FINISH_COUNT = 200;
+    let currentLength = 0;
+
+    for (const {name, data: resource} of names) {
+      try {
+        // Contains Redeems and/or Register
+        const finishes = await this.processNameForFinish(name, resource, acct);
+        const finishesLength = finishes.length;
+        const targetLength = (currentLength + finishesLength);
+
+        if (targetLength > MAX_FINISH_COUNT) {
+          throw new Error(`${name} output length ${finishesLength}`
+             + 'not enough free slots left within transaction');
+        }
+
+        currentLength = targetLength;
+
+        finishes.forEach((element) => {
+          const {output, outpoint} = element;
+          mtx.addOutpoint(outpoint);
```

Comment:
For my understanding, what is the difference between this line and the one below it?

Reply:
outpoint contains the previous output (prevout) (which forms the inputs i believe) where as the output is the outputs that we are creating with this transaction... per my understanding